collect_app/src/main/java/org/odk/collect/android/widgets/BarcodeWidget.java

```diff
@@ -97,35 +95,31 @@ public IAnswerData getAnswer() {
     @Override
     public void setBinaryData(Object answer) {
         String response = (String) answer;
-        stringAnswer.setText(stripInvalidCharacters(response));
-
+        binding.barcodeAnswerText.setText(stripInvalidCharacters(response));
+        binding.barcodeButton.setText(getContext().getString(R.string.replace_barcode));
         widgetValueChanged();
     }
 
     // Remove control characters, invisible characters and unused code points.
     @Contract("null -> null; !null -> !null")
```

Comment:
Please remove this annotation, it's not something common in our codebase and i',m not fan of it.